src/coreclr/jit/fgopt.cpp

```diff
@@ -1947,9 +1948,9 @@ bool Compiler::fgCanCompactBlocks(BasicBlock* block, BasicBlock* bNext)
         return false;
     }
 
-    noway_assert(block->NextIs(bNext));
+    assert(block->NextIs(bNext));
 
-    if (!block->KindIs(BBJ_NONE))
+    if (!block->KindIs(BBJ_ALWAYS) || !block->HasJumpTo(bNext) || (block->bbFlags & BBF_KEEP_BBJ_ALWAYS))
```

Comment:
Coding convention nit: we generally discourage implicit bool conversions (https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/clr-jit-coding-conventions.md#15.1.5).

Reply:
I can add that in a follow-up, since Bruce suggested some other refactors. For now, I'll make the bool conversion explicit.

Reply:
> We could really use a block->HasFlag(...) or similar to make it more obvious what we're doing.

We have a few methods like `BasicBlock::hasProfileWeight` that just check if a flag is set. Would we be interested in replacing these methods with calls to this hypothetical `HasFlag` method?